### ISSUE TITLE
Nayatani_HK.dctl with DR17/ATI/W10

### ISSUE THREAD

**Issue by ConnecTEDDD** [jedypod/open-display-transform#6]
Dear Jed,

Thank you in advance for your work/updates.

The first and the newer Nayatani_HK.dctl are not working in Windows 10 with Resolve 17.2.2 and ATI Radeon VII.

Propably it has a syntax in code that is not supported by ATI GPU.

Gamut Compress 0.7 had a similar issue, but you/Nick have solved it: https://github.com/jedypod/gamut-compress/issues/4

Thank you.

Ted.

**Comment by jedypod** [comment]
Hey @ConnecTEDDD Thanks for opening an issue about this!

After a little digging, it looks like `degrees` and `radians` are built-in functions in OpenCL. I've renamed these in [this commit](https://github.com/jedypod/open-display-transform/commit/acdb07c7076f24fbb994390e811762c94da9343b), which hopefully fixes the issue. What I wouldn't give for tracebacks in DCTL! Debugging is a real pain in this language...

Give it a test and let me know!

**Comment by ConnecTEDDD** [comment]
Hi Jed,

Thank you for taking a look; now it's working.

BTW, BMD had a webinar about DCTL some days.

I have asked the following question to the chat:

''Can you reference the line of code in the error log in case of syntax error? How do you find it a large script?''

Paul Dore replied:

''There is a workaround, albeit a not very practical one.

You subtract a certain value from the line number to find the actual line in the DCTL script.''

**Comment by jedypod** [comment]
Glad to hear it's working! I'll close the issue.

Also, thanks to your comment I have discovered that there is actually an error log! On linux it's in
`<resolve_install_dir>/logs/rollinglog.txt`

I did not know about this before. The DCTL Compilation error doesn't provide the right line number, but it does provide some useful information to track down the issues at least! This will save me a ton of time in the future, so thank you very much! :)
